### PR TITLE
Add RadialFunc and PairwiseConv layers for SE3-Transformer model

### PR DESCRIPTION
**Pull request by JoseAntonioSiguenza**
## Description

Initial implementation of PairwiseConv, BN, and RadialFunc layers for the SE(3)-Transformer model. PairwiseConv ensures SE(3)-equivariant feature interactions, decomposing them into radial (distance-based) and angular (spherical harmonics) components. BN applies per-channel normalization for stability, while RadialFunc learns flexible distance-based interactions with batch normalization.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/layers.py [hunk 1]
```diff
@@ -7302,3 +7302,276 @@ def cosine_dist(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
 
     cosine_similarity = torch.matmul(x_norm, y_norm.transpose(-1, -2))
     return cosine_similarity
+
+
+class BN(nn.Module):
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:7307]: SE3EquivariantLayerNormalization? BN is a bit too short. Feel free to come up with a better name too

File: deepchem/models/torch_models/layers.py [hunk 2]
```diff
@@ -7302,3 +7302,276 @@ def cosine_dist(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
...
+        
+        Returns
+        -------
+        torch.Tensor
+            Normalized tensor with the same shape as input.
+        """
+        return self.bn(x)
+
+
+class RadialFunc(nn.Module):
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:7367]: SE3RadialFunc?

File: deepchem/models/torch_models/layers.py [hunk 3]
```diff
@@ -7302,3 +7302,276 @@ def cosine_dist(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
...
+        -------
+        torch.Tensor
+            Output tensor of shape (-1, out_dim, 1, in_dim, 1, num_freq)
+        """
+        x = x.cuda() if torch.cuda.is_available() else x.cpu()
+        y = self.net(x)
+        return y.view(-1, self.out_dim, 1, self.in_dim, 1, self.num_freq)
+
+
+class PairwiseConv(nn.Module):
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:7455]: SE3PairwiseConv?

**Review by rbharath — COMMENTED** [review]
Generally good, need to rename layers and rebase and should be good to merge after
